torch_geometric/loader/neighbor_loader.py

```diff
@@ -329,28 +436,58 @@ def __repr__(self) -> str:
 ###############################################################################
 
 
-def get_input_nodes(data: Union[Data, HeteroData],
-                    input_nodes: InputNodes) -> Tuple[Optional[str], Sequence]:
+def get_input_nodes(
+    data: Union[Data, HeteroData, Tuple[FeatureStore, GraphStore]],
+    input_nodes: Union[InputNodes, TensorAttr],
+) -> Tuple[Optional[str], Sequence]:
+    def from_bool_tensor(tensor):
+        return tensor.nonzero(as_tuple=False).view(-1)
+
     if isinstance(data, Data):
         if input_nodes is None:
             return None, range(data.num_nodes)
         if input_nodes.dtype == torch.bool:
-            input_nodes = input_nodes.nonzero(as_tuple=False).view(-1)
+            input_nodes = from_bool_tensor(input_nodes)
         return None, input_nodes
 
-    assert input_nodes is not None
+    elif isinstance(data, HeteroData):
+        assert input_nodes is not None
 
-    if isinstance(input_nodes, str):
-        return input_nodes, range(data[input_nodes].num_nodes)
+        if isinstance(input_nodes, str):
+            return input_nodes, range(data[input_nodes].num_nodes)
 
-    assert isinstance(input_nodes, (list, tuple))
-    assert len(input_nodes) == 2
-    assert isinstance(input_nodes[0], str)
+        assert isinstance(input_nodes, (list, tuple))
+        assert len(input_nodes) == 2
+        assert isinstance(input_nodes[0], str)
 
-    if input_nodes[1] is None:
-        return input_nodes[0], range(data[input_nodes[0]].num_nodes)
+        if input_nodes[1] is None:
+            return input_nodes[0], range(data[input_nodes[0]].num_nodes)
 
-    node_type, input_nodes = input_nodes
-    if input_nodes.dtype == torch.bool:
-        input_nodes = input_nodes.nonzero(as_tuple=False).view(-1)
-    return node_type, input_nodes
+        node_type, input_nodes = input_nodes
+        if input_nodes.dtype == torch.bool:
+            input_nodes = from_bool_tensor(input_nodes)
+        return node_type, input_nodes
+
+    else:  # Tuple[FeatureStore, GraphStore]
```

Comment:
This needs to support the same cases as `HeteroData` branch.

Reply:
I think we do:
* `input_nodes` is a `Tensor` or `[str, Tensor]` tuple = supported exactly
* `input_nodes` is a `str` = `input_nodes` is a `TensorAttr` with a `group_name` set (we also require `attr_name` to be set here for num node inferral, but for now I don't see how to get around that)

If there are others, we can include them in a follow-up PR (don't want to overcomplicate this one).

Reply:
Oh yeah, seemed like I misunderstood since we pass in `TensorAttr` it can only hold `bool` tensors. Correct?